eventkitd-bigquery/bigquery/destination.go

```diff
@@ -0,0 +1,67 @@
+package bigquery
+
+import (
+	"context"
+	"fmt"
+	"github.com/jtolio/eventkit"
+	"github.com/jtolio/eventkit/pb"
+	"os"
+	"time"
+)
+
+// BigQueryDestination can be used to save each evenkit package directly to server.
+type BigQueryDestination struct {
+	name           string
+	client         *BigQueryClient
+	SourceInstance string
+	appName        string
+}
+
+var _ eventkit.Destination = &BigQueryDestination{}
+
+func NewBigQueryDestination(ctx context.Context, appName string, project string, dataset string) (*BigQueryDestination, error) {
+	c, err := NewBigQueryClient(ctx, project, dataset)
+	if err != nil {
+		return nil, err
+	}
+
+	res := &BigQueryDestination{
+		client:  c,
+		appName: appName,
+	}
+	host, err := os.Hostname()
+	if err == nil {
+		res.SourceInstance = host
+	}
+	return res, nil
+}
+func (b *BigQueryDestination) Submit(event *eventkit.Event) {
```

Comment:
oh my goodness, synchronously? 

proposal: i think you should have a worker and a buffered channel, and the worker talks to the db, and Submitting just writes to the buffered channel.

if you're worried about event loss, then don't do the thing the udp sender does, which is drop if the buffered channel is full. block then, but that still seems preferable to blocking every time like this.

Reply:
I totally agree. But it might be more useful to put it to `evenkitd-bigquery/bigquery/client.go`. That can be used both by the bq daemon and this destination.

BTW, I planned to use this destination only with evenkit-time, where it's not an issue (one single event) and mainly for debugging missing events.

 But with proper buffering/worker we can consider to use it with satellite-api. (still don't have strong opinion if it's better than UDP or not).